base/compiler/compiler.jl

```diff
@@ -48,6 +48,24 @@ include("operators.jl")
 include("pointer.jl")
 include("refvalue.jl")
 
+# required for bootstrap
+extrema(itr) = extrema(identity, itr)
+function extrema(f, itr)
```

Comment:
Maybe define this as `_extrema_itr` and do `extrema(f, itr) = _extrema_itr(f, itr)`, so that you can test that `_extrema_itr` gives the expected results? Otherwise AFAICT tests won't cover it.

Reply:
This function is in `Core.Compiler` and not in `Base`. So, this won't be tested in the test suite. We can test `Core.Compiler.extrema`, though.

Reply:
I added a few simple tests for `Core.Compiler.extrema` 4a27df97501dc9ee02581480b7076a1ed69e785c